proxy-support/README.md

```diff
@@ -0,0 +1,267 @@
+- Installation: An installation defines the combination of a `MC` with all the provisioned `WCs` from there.
+- App Platform: Refers to a set of features and concepts that allow to browse, install and manage the configurations of apps from the management cluster.
+- Proxy configuration: In most cases a combination of valid `http_proxy`,`https_proxy` and `no_proxy` environment variables.
+
+## Summary
+
+Goal of this proposal is to collect all requirements and information needed to create an installation from scratch behind a proxy server.
+
+## Motivation
+
+Due some security related policies, some customers are enforced to use a corporate proxy server when accessing external resources. Beside some application specific API endpoints this also affects accessing our used:
+
+- container registries (e.g. `quay.io`, `docker.io`)
+- GitOps workflow (e.g. accessing `github.com`)
+
+## Affected Components
+
+### `containerd`
+
+To make `containerd` use of a proxy configuration for pulling container images, it's required to create a `systemd` drop-in and place it under `/etc/systemd/system/containerd.service.d/http-proxy.conf`.
+
+```
+[Service]
+Environment="HTTP_PROXY={{ .HttpProxy }}"
+Environment="http_proxy={{ .HttpProxy }}"
+Environment="HTTPS_PROXY={{ .HttpsProxy }}"
+Environment="https_proxy={{ .HttpsProxy }}"
+Environment="NO_PROXY={{ .NoProxy}}"
+Environment="no_proxy={{ .NoProxy }}"
+```
+
+### workload on top of Kubernetes
+
+Every pod within a Kubernetes cluster needs a valid proxy configuration if the pod needs to access external systems.
+
+On a `MC` there are a lot of components which needs access to external systems:
+
+- `Flux` needs access to `github.com`
+- The entire monitoring and alerting stack needs access to `opsgenie`, `grafana`, ...
+- App platform components need access to an OCI artifact system (e.g. `github.com`, `azure`)
+
+As we already use `kyverno` in our `MC`, we can use the [`kyverno-policies-connectivity`] chart to inject the proxy configuration into all pods.
+
+> Please note: 
+> - due multiple valid reasons, `kyverno` doesn't mutate objects in `namespace/kube-system` per default.
+> - static pods are managed by the `kubelet`. For these pods it's not possible to use `kyverno` for mutation. Therefore it's required to inject the proxy configuration during the machine creation phase (could be achieved in `CAPI` based clusters via `kubeadm`)
+
+#### special workload
+
+There are some components, which are part of the default set on top of a cluster. These components can't get configured via `kyverno` as they are either in `namespace/kube-system` (for `cert-manager`) or must exist before we apply applications via the app platform (for `chart-operator`).
+
+##### `cert-manager`
+
+As `cert-manager` is needed in every cluster (`MC` and `WC`) and it's running in `namespace/kube-system`. `cert-manager` interact with external `ACME` services (`lets encrypt` at the moment).
```

Comment:
Did kyverno not move to the kyverno namespace? Is there still a reason preventing it from running in the kube-system namespace?

Reply:
Kyverno has moved and is now running in the `kyverno` namespace by default. 

However, to support cases where customers run very small, single-master clusters, product decided to exclude `kube-system` by default so that the cluster still can recover from the loss of all nodes. The intent is currently to re-enable Kyverno in `kube-system` once we've found a way to still support recovery from total node loss. Here is an issue from the implementation which references the product discussion https://github.com/giantswarm/giantswarm/issues/24707

Reply:
That is sad news :( Do we have some documentation explaining how to configure it in the pod spec then?

Reply:
Configure the proxy? Or configure Kyverno to also apply in `kube-system`?

Reply:
Sorry, I meant the proxy :)

Reply:
will create an issue and prepare an initial documentation